File: packages/common/erp-common-expressjs/src/middlewares/authentication.middleware.ts
import { Request, Response, NextFunction } from 'express';
import { IAuth } from '@gmahechas/erp-common';
import { Context, jwtDecode, sendError, TypeErrorMessage, uuidv4, Winston } from '@gmahechas/erp-common-ms-utils-js';

export const authenticationMiddleware = async (request: Request, response: Response, next: NextFunction) => {
	const requestId = request.headers['x-request-id'] as string || uuidv4();
	Context.set('requestId', requestId);
	const { authorization: token } = request.headers;
	const { body, originalUrl, method, params, query } = request;
	if (!token) {
		Winston.logger.error(TypeErrorMessage.AUTHENTICATION, { requestId: Context.get('requestId'), auth: JSON.stringify(null), action: originalUrl, method, payload: JSON.stringify({ body, params, query }) });
		Context.clear();
		sendError(TypeErrorMessage.AUTHENTICATION);
	}
	const auth = jwtDecode(token) as IAuth;
	Context.set('token', token);
	Context.set('auth', auth);
	next();
};

File: packages/common/erp-common-flaskpy/gmahechas_erp_common_flaskpy/middlewares/authentication_middleware.py
from functools import wraps
from flask import request, Response
from gmahechas_erp_common_py import response
from gmahechas_erp_common_ms_utils_py import JWT, Context


def authentication_middleware(function):
    @wraps(function)
    def wrapper(*args, **kwargs):
        token = request.headers.get("Authorization")
        if token is None:
            data = response(400, error={"type": "authentication"})
            status_code, body = data['statusCode'], data['body']
            return Response(body, mimetype='application/json', status=status_code)

        auth = JWT.decode(token)
        Context.set('token', token)
        Context.set('auth', auth)
        return function(*args, **kwargs)

    return wrapper
